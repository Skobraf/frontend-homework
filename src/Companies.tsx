import { connect } from "react-redux";
import { createStructuredSelector } from "reselect";

import { ReduxState, Company } from "./types";
import { getCompanies, getSingleCompany } from "./selectors";
import { setSelectedCompanyId, toggleDropdownMenuVisibility } from "./actions";

import CompanyLink from "./CompanyLink";

type ReduxProps = {
  companies: Array<Company>;
  singleCompany:Company[];
};

type DispatchProps = {
  setSelectedCompanyId:(id:number) => void;
  toggleDropdownMenuVisibility: () => void;
};

export const Companies = ({ companies,  setSelectedCompanyId, toggleDropdownMenuVisibility, singleCompany }: ReduxProps & DispatchProps) => (
  <>
    <div className="list-item" onClick={toggleDropdownMenuVisibility}>Your companies</div>

    {companies.map((company) => (
      <CompanyLink
        key={company.id}
        {...company}
        setSelectedCompanyId={setSelectedCompanyId}
        toggleDropdownMenuVisibility={toggleDropdownMenuVisibility}
        singleCompany={singleCompany}
      />
    ))}
  </>
);

export default connect(
  createStructuredSelector<ReduxState, ReduxProps>({
    companies: getCompanies,
    singleCompany:getSingleCompany
  }),
  { setSelectedCompanyId, toggleDropdownMenuVisibility }
)(Companies);
