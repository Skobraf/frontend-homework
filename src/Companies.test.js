import { shallow } from 'enzyme'
import { Companies } from './Companies'

describe('<Companies />', () => {
  const companies = [
    { id: 1, name: 'Dummy company' },
    { id: 2, name: 'Smarty company' },
  ]
  const singleCompany = [ {name:'first company'}]

  it('renders list of company links', () => {
    expect(shallow(<Companies companies={companies} singleCompany={singleCompany} />)).toMatchSnapshot()
  })
})
