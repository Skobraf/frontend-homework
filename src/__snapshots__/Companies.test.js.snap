// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Companies /> renders list of company links 1`] = `
<Fragment>
  <div>
    Your companies
  </div>
  <CompanyLink
    id={1}
    key="1"
    name="Dummy company"
  />
  <CompanyLink
    id={2}
    key="2"
    name="Smarty company"
  />
</Fragment>
`;
